src/Microsoft.VisualStudio.ProjectSystem.Managed/Build/BuildUtilities.cs

```diff
@@ -66,21 +66,16 @@ public static ProjectPropertyElement GetProperty(ProjectRootElement project, str
         /// <param name="propertyValue">Value of the property to evaluate.</param>
         /// <param name="delimiter">Character used to delimit the property values.</param>
         /// <returns>Collection of individual values in the property.</returns>
-        public static ImmutableArray<string> GetPropertyValues(string propertyValue, char delimiter = ';')
+        public static IEnumerable<string> GetPropertyValues(string propertyValue, char delimiter = ';')
         {
-            IEnumerable<string> values = propertyValue.Split(delimiter).Select(f => f.Trim());
-
             // We need to ensure that we return values in the specified order.
-            ImmutableArray<string>.Builder valuesBuilder = ImmutableArray.CreateBuilder<string>();
-            foreach (string value in values)
+            foreach (string value in new LazyStringSplit(propertyValue, delimiter))
             {
-                if (!string.IsNullOrEmpty(value))
+                if (!string.IsNullOrWhiteSpace(value))
                 {
-                    valuesBuilder.Add(value);
+                    yield return value.Trim();
                 }
             }
-
-            return valuesBuilder.Distinct(StringComparer.OrdinalIgnoreCase).ToImmutableArray();
```

Comment:
This change might not be correct actually. The previous code eliminated duplicates. Will review.

Reply:
I restored deduplication and added a test with duplicates.